server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -1037,4 +1048,7 @@ def generate_token(
         batch.prefill_next_token_indices = None
         batch.max_seqlen = batch.max_seqlen + 1
 
+        batch.kv_cache_usage = CACHE_MANAGER.usage()
```

Comment:
I'm a bit worried that this call might cause perf drop. Can you try with and without it.